x-pack/plugin/autoscaling/src/test/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderDecisionTests.java

```diff
@@ -692,8 +697,7 @@ static DiscoveryNode newDataNode(DiscoveryNodeRole role, String nodeName) {
     }
 
     private static String randomNodeId(RoutingNodes routingNodes, DiscoveryNodeRole role) {
-        return randomFrom(routingNodes.stream().map(RoutingNode::node).filter(n -> n.getRoles().contains(role)).collect(Collectors.toSet()))
-            .getId();
+        return randomValueOtherThanMany(not(node -> node.node().getRoles().contains(role)), () -> randomFrom(routingNodes)).nodeId();
```

Comment:
Maybe I'm misunderstanding this, but why is this change needed? This seems to be not sampling from the same possible set of node IDs compared to the previous code.

Reply:
This is not required change, I tried to refactor picking the node to not collect to set.
I find `many` confusing in `randomValueOtherThanMany`, this code is keep trying to pick a random node from `routingNodes` (with second argument: `() -> randomFrom(routingNodes)`) for as long as it does not contain requested role (first predicate argument).

I could revert it if you believe this is harder to read then before.

Reply:
Well, it's up to you. But I find the `filter` variant more straight-forward.